### PR TITLE
[Merged by Bors] - Complete example for how k8sSearch works

### LINKED ISSUE #273: Document that k8sSearch secrets must be labeled with the secretclass
Currently our docs (https://docs.stackable.tech/home/stable/secret-operator/secretclass.html#backend-k8ssearch) only mention the scope labels, the class label is also required: https://github.com/stackabletech/secret-operator/blob/07540604e01fdb13962602280c66c965886c1a39/rust/operator-binary/src/backend/k8s_search.rs#L121-L122

### PR DESCRIPTION
**Pull request by razvan**
Fixes #273

### CODE REVIEW

File: docs/modules/secret-operator/pages/secretclass.adoc [hunk 1]
```diff
@@ -119,8 +119,61 @@ spec:
...
-to one file. It is suggested these `Secret` objects should follow one of the xref:#format[formats defined in this document].
+This backend can be used to mount `Secret` across namespaces into pods. The `Secret` object is selected based on two things:
+
+1. The xref:scope.adoc[scopes] specified on the `Volume` using the attribute `secrets.stackable.tech/scope`.
+2. The label `secrets.stackable.tech/class` specified in the `Secret` it's self. This must match the name of the `SecretClass`.
+
+Each field in this `Secret` is mapped to one file. It is suggested these `Secret` objects should follow one of the xref:#format[formats defined in this document].
+
+In the example below, given the three object definitions for a `Pod`, a `SecretClass` and a `Secret`, the operator will first read the Pod's volume attributes then look up the secret class. The `k8sSearch` backend will look up the secret object labeled with `secrets.stackable.tech/class: admin-credentials-class` (the name of the secret class) and mount the fields of the secret as files into the the container at the specified mount point (`/credentials`).
```
Comment (sbernauer) [inline docs/modules/secret-operator/pages/secretclass.adoc:129]:
```suggestion
In the example below, given the three object definitions for a `Pod`, a `SecretClass` and a `Secret`, the operator will first read the Pod's volume attributes then look up the secret class. The `k8sSearch` backend will look up the Secret object labeled with `secrets.stackable.tech/class: admin-credentials-class` (the name of the secret class) and mount the fields of the Secret as files into the container at the specified mount point (`/credentials`).
Please note that the contents in the volume will not update when the Secret content changes. A Pod restart is needed to refresh the Secret contents on disk.
```

File: docs/modules/secret-operator/pages/secretclass.adoc [hunk 2]
```diff
@@ -119,8 +119,61 @@ spec:
...
+    volumeMounts:
+    - name: admin-credentials-volume
+      mountPath: /credentials
+  volumes:
+  - name: admin-credentials-volume
+    csi:
+      driver: secrets.stackable.tech
+      volumeAttributes:
+        secrets.stackable.tech/class: admin-credentials-class
+        secrets.stackable.tech/scope: pod
```
Comment (sbernauer) [inline docs/modules/secret-operator/pages/secretclass.adoc:154]: CSI is deprecated (see https://github.com/stackabletech/secret-operator/pull/125)
```suggestion
  - name: admin-credentials-volume
    ephemeral:
      volumeClaimTemplate:
        metadata:
          annotations:
            secrets.stackable.tech/class: admin-credentials-class
            secrets.stackable.tech/scope: pod
        spec:
          storageClassName: secrets.stackable.tech
          accessModes:
            - ReadWriteOnce
          resources:
            requests:
              storage: "1"
```

File: docs/modules/secret-operator/pages/secretclass.adoc [hunk 3]
```diff
@@ -119,8 +119,61 @@ spec:
...
+spec:
+  backend:
+   k8sSearch:
+      searchNamespace:
+        pod: {}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: superset-with-ldap-server-veri-tls-ldap-bind
```
Comment (sbernauer) [inline docs/modules/secret-operator/pages/secretclass.adoc:169]:
```suggestion
  name: my-admin-credentials
```

**Review by sbernauer — APPROVED** [review]
Thx! LGTM % comments

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**